File: golang/user_assets.go
package ccpayment

import "context"

type UserAssetsService struct {
	client *Client
}

// GetUserCoinAssetList gets all token assets of a specified user
func (s *UserAssetsService) GetUserCoinAssetList(ctx context.Context, userId string) (*GetUserCoinAssetListResponse, error) {
	req := &GetUserCoinAssetListRequest{UserId: userId}
	var result GetUserCoinAssetListResponse
	err := s.client.post(ctx, "/getUserCoinAssetList", req, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

// GetUserCoinAsset gets a specific token asset of a specified user
func (s *UserAssetsService) GetUserCoinAsset(ctx context.Context, userId string, coinId uint64) (*GetUserCoinAssetResponse, error) {
	req := &GetUserCoinAssetRequest{UserId: userId, CoinId: coinId}
	var result GetUserCoinAssetResponse
	err := s.client.post(ctx, "/getUserCoinAsset", req, &result)
	if err != nil {
		return nil, err
	}
	return &result, nil
}

File: python/ccpayment/services/user_assets.py
"""User Assets Service"""


class UserAssetsService:
    def __init__(self, client):
        self.client = client

    def get_user_coin_asset_list(self, user_id: str) -> dict:
        """Get all token assets of a specified user"""
        result = self.client._post("/getUserCoinAssetList", {"userId": user_id})
        return result

    def get_user_coin_asset(self, user_id: str, coin_id: int) -> dict:
        """Get a specific token asset of a specified user"""
        result = self.client._post("/getUserCoinAsset", {
            "userId": user_id,
            "coinId": coin_id
        })
        return result.get("asset", {})
